### PR TITLE
Add clink utils

### PR DESCRIPTION
**Pull request by rashil2000**
**Clink**

[Clink](https://chrisant996.github.io/clink/) is an extension framework which incorporates GNU Readline style editing in cmd.exe. It is currently internally used by the popular terminal package [Cmder](https://cmder.net/). The manifest for Clink is already present in Main - https://github.com/ScoopInstaller/Main/blob/master/bucket/clink.json

**Clink Completions**

This provides programmatic completions for various external programs leveraging the Clink framework. This is also used internally by Cmder.

**Clink Flex Prompt**

This is a Clink port of the popular Zsh prompt tool [Powerlevel10k](https://github.com/romkatv/powerlevel10k). This project is quite new, but since it is made by the author of Clink itself, I consider it trusted.

### CODE REVIEW

File: bucket/clink-completions.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    "suggest": {
+        "Clink": "clink"
+    },
+    "url": "https://github.com/vladimir-kotikov/clink-completions/archive/0.3.7.zip",
+    "hash": "2136a9b868fe05af7fb07a3b8a3c878b551bf6601321c91a6390376b47c1fad6",
+    "extract_dir": "clink-completions-0.3.7",
+    "installer": {
+        "script": [
+            "if (Get-Command clink -ErrorAction SilentlyContinue) {",
+            "   clink installscripts $dir",
```
Comment (niheaven) [inline bucket/clink-completions.json:15]:
```suggestion
            "   clink installscripts \"$dir\"",
```

File: bucket/clink-completions.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            "} else {",
+            "   \"Clink installation not found. Please manually install these scripts.\"",
```
Comment (niheaven) [inline bucket/clink-completions.json:17]:
```suggestion
            "   warn 'Clink installation not found. Please manually install these scripts.'",
```

File: bucket/clink-completions.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            "}"
+        ]
+    },
+    "uninstaller": {
+        "script": [
+            "if (Get-Command clink -ErrorAction SilentlyContinue) {",
+            "   clink uninstallscripts $dir",
```
Comment (niheaven) [inline bucket/clink-completions.json:24]:
```suggestion
            "   clink uninstallscripts \"$dir\"",
```

File: bucket/clink-flex-prompt.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "license": "MIT",
+    "suggest": {
+        "Clink": "clink"
+    },
+    "url": "https://github.com/chrisant996/clink-flex-prompt/releases/download/v0.5/clink-flex-prompt-0.5.zip",
+    "hash": "16e0e8a58d9f12acb9e29deafd258f033adb9d7ecaa81f9ea43a952827b298c6",
+    "installer": {
+        "script": [
+            "if (Get-Command clink -ErrorAction SilentlyContinue) {",
+            "   clink installscripts $dir",
```
Comment (niheaven) [inline bucket/clink-flex-prompt.json:14]:
```suggestion
            "   clink installscripts \"$dir\"",
```

File: bucket/clink-flex-prompt.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+            "} elseif ($Env:CMDER_ROOT) {",
+            "   & $Env:CMDER_ROOT\\vendor\\clink\\clink.bat installscripts $dir",
```
Comment (niheaven) [inline bucket/clink-flex-prompt.json:16]:
```suggestion
            "   & \"$Env:CMDER_ROOT\\vendor\\clink\\clink.bat\" installscripts \"$dir\"",
```

File: bucket/clink-flex-prompt.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            "} else {",
+            "   \"Clink or Cmder installation not found. Please manually install these scripts.\"",
```
Comment (niheaven) [inline bucket/clink-flex-prompt.json:18]:
```suggestion
            "   warn 'Clink or Cmder installation not found. Please manually install these scripts.'",
```

File: bucket/clink-flex-prompt.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            "}"
+        ]
+    },
+    "uninstaller": {
+        "script": [
+            "if (Get-Command clink -ErrorAction SilentlyContinue) {",
+            "   clink uninstallscripts $dir",
```
Comment (niheaven) [inline bucket/clink-flex-prompt.json:25]:
```suggestion
            "   clink uninstallscripts \"$dir\"",
```

File: bucket/clink-flex-prompt.json [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            "} elseif ($Env:CMDER_ROOT) {",
+            "   & $Env:CMDER_ROOT\\vendor\\clink\\clink.bat uninstallscripts $dir",
```
Comment (niheaven) [inline bucket/clink-flex-prompt.json:27]:
```suggestion
            "   & \"$Env:CMDER_ROOT\\vendor\\clink\\clink.bat\" uninstallscripts \"$dir\"",
```

**Review by niheaven — CHANGES_REQUESTED** [review]
Nest `$dir` with `"` to make sure dir w/ whitespace should work.

**Comment by rashil2000** [comment]
Done

**Review by niheaven — APPROVED** [review]
LGTM

**Pull request merged**